src/pages/signin/ChooseSSOOrMagicCode.js

```diff
@@ -79,21 +70,12 @@ function ChooseSSOOrMagicCode({credentials, account, setIsUsingMagicCode}) {
                     </Text>
                 </View>
 
-                <Button
-                    isDisabled={isOffline}
-                    style={[styles.mv3]}
-                    text={translate('samlSignIn.useMagicCode')}
-                    isLoading={account.isLoading && account.loadingForm === (account.requiresTwoFactorAuth ? CONST.FORMS.VALIDATE_TFA_CODE_FORM : CONST.FORMS.VALIDATE_CODE_FORM)}
-                    onPress={() => {
-                        Session.resendValidateCode(credentials.login);
```

Comment:
I'm actually started to thinking about this line - thats where we are requesting new magic code. Removing this line should be sufficient for removing the double code problem - what do you think?

Reply:
Oh, that was my first try, the thing here is:
1- The user goes first to SignInPage, puts his email, and goes to step two (ChooseSSOORMagicCode component), magic code is sent
2- The user sees the SingleSignOn input and then sees the button for putting the code but he doesn't know he received one magic code 
2.1 In an alternative scenario where the user knows it, he should be going running before the time's up and writing the code clicking the button to activate the magic code form.